### PR TITLE
Core/Spell: Some non-damage spells should break stealth

### PR DESCRIPTION
**Pull request by pjasicek**
Source:

http://www.wowhead.com/forums&topic=124541.2

### CODE REVIEW

**Comment by MitchesD** [comment]
doesn't it affects more spells than intended? I wonder if it can break something

**Comment by robinsch** [comment]
https://github.com/TrinityCore/TrinityCore/issues/14819

File: src/server/game/Spells/SpellMgr.cpp [hunk 1]
```diff
@@ -2821,6 +2822,10 @@ void SpellMgr::LoadSpellInfoCustomAttributes()
                 case SPELL_AURA_POWER_BURN:
                     spellInfo->AttributesCu |= SPELL_ATTR0_CU_NO_INITIAL_THREAT;
                     break;
+                case SPELL_AURA_PERIODIC_DAMAGE:        // DoTs like Shadow Word: Pain
+                    spellInfo->AttributesCu |= SPELL_ATTR0_CU_NO_INITIAL_THREAT;
```
Comment (Shauren) [inline src/server/game/Spells/SpellMgr.cpp]: whaaaaat

Comment (pjasicek) [inline src/server/game/Spells/SpellMgr.cpp]: What do you mean ? This flag is assigned on master, I did not make this up. I only added the SPELL_ATTR0_CU_AURA_CC flag.

Comment (DDuarte) [inline src/server/game/Spells/SpellMgr.cpp]: If this code was on master (actually `3.3.5`) the diff wouldn't show these added lines.

https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Spells/SpellMgr.cpp#L2821

Comment (pjasicek) [inline src/server/game/Spells/SpellMgr.cpp]: Please look at this line on master:

https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Spells/SpellMgr.cpp#L2813

In my commit I only moved the "case SPELL_AURA_PERIODIC_DAMAGE:" so I would be able to add the SPELL_ATTR0_CU_AURA_CC flag.

**Comment by robinsch** [comment]
Still this is a incorrect fix, any harmful spell should break stealth.

File: src/server/game/Spells/SpellMgr.cpp (hunk 1 above)
Comment (DDuarte) [inline src/server/game/Spells/SpellMgr.cpp]: You are correct, I'm sorry.

**Comment by jackpoz** [comment]
@pjasicek which spells does this PR affect ?

**Comment by Aokromes** [comment]
Any new on this?

**Comment by tkrokli** [comment]
@pjasicek's account page does not show any activity after 2015-05-31.

**Comment by jackpoz** [comment]
this PR modifies same lines as https://github.com/TrinityCore/TrinityCore/pull/14824

**Pull request closed without merging**